### PR TITLE
On-the-fly RICH PID and tracking corrections

### PR DESCRIPTION
**Pull request by NNicassio99**

### CODE REVIEW

**Comment by hscheid** [comment]
Hi @NNicassio99 it seems you reverted some of the changes that @ddobrigk and I implemented in the TrackSmearer. Could you check that again? It looks like the interpolation of the efficiency was removed as well as a small change I implemented in the efficiency return values.

File: ALICE3/Core/DelphesO2TrackSmearer.cxx [hunk 1]
```diff
@@ -181,19 +181,19 @@ bool TrackSmearer::smearTrack(O2Track& o2track, int pid, float nch)
 // uncertainty on pt
 double TrackSmearer::getPtRes(int pdg, float nch, float eta, float pt)
 {
-  auto lutEntry = getLUTEntry(pdg, nch, 0., eta, pt);              // take variance 1/pt
-  auto ptRes = sqrt(lutEntry->covm[14]) * pow(lutEntry->pt, 2);    // propagate to pt uncertainty
-  //ptRes /= lutEntry->pt;                                         // relative uncertainty
+  auto lutEntry = getLUTEntry(pdg, nch, 0., eta, pt);           // take variance 1/pt
+  auto ptRes = sqrt(lutEntry->covm[14]) * pow(lutEntry->pt, 2); // propagate to pt uncertainty
```
Comment (njacazio) [inline ALICE3/Core/DelphesO2TrackSmearer.cxx:185]: Hi @NNicassio99 I guess these can stay unchanged

**Comment by NNicassio99** [comment]
Hi @hscheid,

yes, you are right, I will revise the pull request or create a new one

**Comment by vkucera** [comment]
> Hi @hscheid,
> 
> yes, you are right, I will revise the pull request or create a new one

Please do not create a new PR! Just push updates to your branch.

**Pull request closed without merging**